bool stop
float32 rotation_angle
